base/mysqlxx/Exception.h

```diff
@@ -26,6 +26,15 @@ struct ConnectionFailed : public Exception
 };
 
 
+/// Connection to MySQL server was lost
+struct ConnectionLost : public Exception
+{
+    ConnectionLost(const std::string & msg, int code = 0) : Exception(msg, code) {}
```

Comment:
`using Exception::Exception` seem to be better, cause it will allow to use other constructors.